app/lib/features/home/providers/notifiers/always_client_notifier.dart

```diff
@@ -9,26 +9,29 @@ class AlwaysClientNotifier extends AsyncNotifier<Client> {
   late ProviderSubscription subscription;
   @override
   FutureOr<Client> build() async {
-    subscription = ref.listen<AsyncValue<Client?>>(clientProvider, _updated);
-    final client = await ref.read(clientProvider.future);
-    if (client != null) {
-      return client;
-    }
-
     final cpl = completer ??= Completer<Client>();
+    subscription = ref.listen<AsyncValue<Client?>>(
+      clientProvider,
+      _updated,
+      fireImmediately: true,
+    );
     return cpl.future;
```

Comment:
I think destruction code is needed.
```dart
ref.onDispose(() => subscription.close());
```

Reply:
not really. it will be dropped and the drop will cause a close.